common/models/user.js

```diff
@@ -247,6 +247,12 @@ module.exports = function(User) {
             debug('An error is reported from User.hasPassword: %j', err);
             fn(defaultError);
           } else if (isMatch) {
+            if ((credentials.password).length > MAX_PASSWORD_LENGTH) {
+              err = new Error(g.f('login failed as the password entered is too long'));
```

Comment:
Please use `defaultError` here, so that we don't disclose any extra information that could be potentially used by an attacker.